shared/login/join-or-login/index.tsx

```diff
@@ -24,43 +27,71 @@ const Feedback = ({onFeedback}) =>
     </Kb.Text>
   )
 
-const Intro = (props: Props) => (
-  <Kb.Box2 direction="vertical" fullWidth={true} fullHeight={true}>
-    {!!props.bannerMessage && (
-      <Kb.Box2 direction="vertical" fullWidth={true} style={styles.banner}>
-        <Kb.Text center={true} type="BodySmallSemibold" style={styles.bannerMessage}>
-          {props.bannerMessage}
-        </Kb.Text>
-      </Kb.Box2>
-    )}
-    <Kb.Box2
-      direction="vertical"
-      fullWidth={true}
-      fullHeight={true}
-      gap="large"
-      style={styles.innerContainer}
-    >
-      <Kb.Box2 direction="vertical" fullWidth={true} gap="small" style={styles.alignItemsCenter}>
-        <Kb.Icon type="icon-keybase-logo-80" />
-        <Kb.Text type="HeaderBig" style={styles.join}>
-          Join Keybase
-        </Kb.Text>
-        <Kb.ButtonBar>
-          <Kb.Button onClick={props.onSignup} label="Create an account" />
-        </Kb.ButtonBar>
-      </Kb.Box2>
-      <Kb.Box2 direction="vertical" fullWidth={true} gap="tiny" style={styles.alignItemsCenter}>
-        <Kb.Text type="Body" onClick={props.onLogin}>
-          Already on Keybase?
-        </Kb.Text>
-        <Kb.ButtonBar>
-          <Kb.Button type="Dim" onClick={props.onLogin} label="Log in" />
-        </Kb.ButtonBar>
+class Intro extends React.Component<Props, {}> {
+  intervalId: NodeJS.Timeout | null
+
+  constructor(props: Props) {
+    super(props)
+
+    this.intervalId = null
+  }
+
+  componentDidMount() {
+    this.props.checkIsOnline()
+    this.intervalId = setInterval(this.props.checkIsOnline, 2000)
```

Comment:
Ah -- I think we should use the version of setInterval that's inside common-adapters/hoc-timers, using the native one can crash if the component's not mounted.

Reply:
Done.